src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -68,20 +68,39 @@ class TransportManagementService implements Serializable {
 
         def proxy = config.proxy ? config.proxy : script.env.HTTP_PROXY
 
-        script.sh """#!/bin/sh -e
-                curl ${proxy ? '--proxy ' + proxy + ' ' : ''} -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o responseFileUpload.txt  --fail '${url}/v2/files/upload'
+        def responseFileUpload = 'responseFileUpload.txt'
+
+        def responseContent
+
+        try {
+            script.sh """#!/bin/sh -e
+                curl ${proxy ? '--proxy ' + proxy + ' ' : ''} -H 'Authorization: Bearer ${token}' -F 'file=@${file}' -F 'namedUser=${namedUser}' -o ${responseFileUpload}  --fail '${url}/v2/files/upload'
             """
 
-        def responseContent = script.readFile("responseFileUpload.txt")
+            echo("File upload successful.")
 
-        if (config.verbose) {
-            echo("${responseContent}")
-        }
+        } catch (Exception e) {
+            String message = 'Exception caught during file upload.'
+            if (config.verbose) {
+                message += ' Consider re-running in verbose mode in order to get more details.'
+            }
+            echo(message)
+            throw e
+        } finally {
 
-        echo("File upload successful.")
+            if (script.fileExists(responseFileUpload)) {
+                responseContent = script.readFile(responseFileUpload)
```

Comment:
What is the reason that file content is read here in the finally section? Why not to read it already in the try section after successful execution of the curl command? And also return already in the try section.

Reply:
Since we are also interested in the body content in case of a failure. E.g. a http 400 (bad request) can have a response body which is supposed to contain valuable hints. In that case `sh` returns with an exception, hence any line behind will not be executed - and we can't do it there. But the finally block is executed anyway. That is the reason why it is in the finally block.

Reply:
... wait: I saw the `--fail`.

I just checked that. In case `--fail` is set the response is omitted. I guess we have to change that ... In case we dont fail we get the response, but we have to check the http status code. Maybe there is a way to configure `curl` to write the response  also in conjunction with `--fail`.

Reply:
Good point!

Reply:
That discussion here is obsolete when we have the other approch with curl --write-out instead of curl --fail.